### DISCUSSION TITLE
`Bitmap` as `Control`

### DISCUSSION THREAD

**Discussion by visualizor** [picoe/Eto#2596] [General]
When I instantiate a `Bitmap` like so:
```
Bitmap pic = new Bitmap("D:\\Myfolder\\mypic.png");
DynamicLayout dy = new DynamicLayout();
dy.Add(pic, True, True);
```
It can be added to the layout even though the `DynamicLayout.Add` method takes a `Control` as its first arg. How does that work?
If the pic gets passed into another method as an `object`, where `Control`s are added to a layout, I only have an `if` to catch `Control`s.
```
if (incoming is Control c){
...
}
```
This `Bitmap` will slip through...

**Comment by cwensley** [comment]
It works because of [this implicit conversion](https://github.com/picoe/Eto/blob/develop/src/Eto/Forms/Controls/Control.cs#L1467) on `Control`.  It works the same for strings to create a `Label`.    It creates an `ImageView` to host the bitmap, so your if statement would return an `ImageView` containing that `Bitmap`.

**Reply by visualizor** [reply]
But i meant the `if` didn't work
I think you're familiar with Rhino3d so here's the issue:
![image](https://github.com/picoe/Eto/assets/25107288/3a0cb0dd-8609-4bcd-a882-9060dec6e5ca)

How can my method catch everything that can implicitly be converted to a `Control` in its args?

**Reply by cwensley** [reply]
Unfortunately, since the incoming parameter is an object (not a Bitmap) there's no way for the compiler to know that it should use an implicit conversion. So if you want your 2nd script to handle a Bitmap/string input but translate it to an Eto control, you have to do it explicitly:

```c#
if (x is Bitmap b)
  x = new ImageView { Image = b };
if (x is string s)
  x = new Label { Text = s };

A = x is Control;
```
